lib/http-bridge/test/integration/Test/Integration/Scenario/CLISpec.hs

```diff
@@ -31,72 +31,167 @@ import Test.Integration.Framework.DSL
+
+    describe "CLI1 - Can create wallet with different mnemonic snd factor sizes" $ do
+        forM_ ["9", "12"] $ \(size) -> it size $ \_ -> do
+            let walletName = "Wallet created via CLI"
+            Stdout mnemonics <- generateMnemonicsViaCLI []
+            Stdout mnemonics2 <- generateMnemonicsViaCLI ["--size", size]
+            let passphrase = "Secure passphrase"
+
+            c <- createWalletViaCLI [walletName] mnemonics mnemonics2 passphrase
+            c `shouldBe` ExitSuccess
+
+            Stdout outList <- listWalletsViaCLI
+            outList `shouldContain` walletName
+
+    describe "CLI1 - Can't create wallet with wrong size of mnemonic" $ do
+        forM_ ["9", "12"] $ \(size) -> it size $ \_ -> do
+            let walletName = "Wallet created via CLI"
+            Stdout mnemonics <- generateMnemonicsViaCLI ["--size", size]
+            Stdout mnemonics2 <- generateMnemonicsViaCLI ["--size", size]
+            let passphrase = "Secure passphrase"
+
+            c <- createWalletViaCLI [walletName] mnemonics mnemonics2 passphrase
+            c `shouldBe` (ExitFailure 1)
+
+            Stdout outList <- listWalletsViaCLI
+            outList `shouldNotContain` walletName
+
+    describe "CLI1 - Can't create wallet with wrong size of mnemonic snd factor" $ do
+        forM_ ["15", "18", "21", "24"] $ \(size) -> it size $ \_ -> do
+            let walletName = "Wallet created via CLI"
+            Stdout mnemonics <- generateMnemonicsViaCLI ["--size", size]
+            Stdout mnemonics2 <- generateMnemonicsViaCLI ["--size", size]
+            let passphrase = "Secure passphrase"
+
+            c <- createWalletViaCLI [walletName] mnemonics mnemonics2 passphrase
+            c `shouldBe` (ExitFailure 1)
+
+            Stdout outList <- listWalletsViaCLI
+            outList `shouldNotContain` walletName
+
     it "CLI - Can get a wallet" $ \ctx -> do
         walId <- createWallet ctx "1st CLI Wallet" mnemonics15
-        (Exit c, Stdout out, Stderr err) <- command [] "cardano-wallet"
-            ["wallet", "get", "--port", "1337", walId ]
+        (Exit c, Stdout out, Stderr err) <- getWalletViaCLI walId
         err `shouldBe` "Ok.\n"
         expectValidJSON (Proxy @ApiWallet) out
         out `shouldContain` "1st CLI Wallet"
         c `shouldBe` ExitSuccess
 
+    describe "CLI - Cannot get wallets with false ids" $ do
+        forM_ falseWalletIds $ \(title, walId) -> it title $ \_ -> do
+            (Exit c, Stdout out, Stderr err) <- getWalletViaCLI walId
+            out `shouldBe` ""
+            if (title == "40 chars hex") then
+                err `shouldBe` "Wallet not found.\n"
+            else
+                err `shouldBe` "wallet id should be an hex-encoded string of\
+                    \ 40 characters\n"
+
+            if (title == "40 chars hex") then
+                c `shouldBe` ExitSuccess
+            else
+                c `shouldBe` ExitFailure 1
```

Comment:
This is slightly inconsistent I think. I suppose the response code should be always `1` (currently for valid hex, but not existing wallet it is `0`, for others it is `1`)

Reply:
Hmmm... That's actually right.